proxy/http/HttpTransact.cc

```diff
@@ -669,6 +669,11 @@ HttpTransact::StartRemapRequest(State *s)
 
     s->hdr_info.client_request.set_url_target_from_host_field();
 
+    // Since we're not doing remap, we still have to allow for these overridable
+    // configurations to modify follow-redirect behavior. Someone could for example
+    // have set them in a plugin other than conf_remap running in a prior hook.
+    s->state_machine->enable_redirection = (s->txn_conf->number_of_redirections > 0);
```

Comment:
Also, since you made me look, I found a possible bad behavior in the plugin APIs (and an old API that's been deprecated since 5.x, which I removed). The APIs will now also enforce a minimum retry count of "1" when you explicitly enable follow redirect or a follow redirect URL.